crates/ruma-macros/src/events/event_enum.rs

```diff
@@ -328,6 +328,28 @@ fn expand_content_enum(
     let serialize_custom_event_error_path =
         quote! { #ruma_events::serialize_custom_event_error }.to_string();
 
+    let serde_json = quote! { #ruma_events::exports::serde_json };
+    let event_type_match_arms: TokenStream = events
+        .iter()
+        .map(|event| {
```

Comment:
This needs to handle any type that ends with `.*` differently, to use `EventContentFromType::from_parts` instead of the `Deserialize` implementation.

It actually made me realize by looking at the code that we should already handle something like that to deserialize `AnyGlobalAccountDataEvent`: https://github.com/ruma/ruma/issues/1851.

Reply:
You can look at #1852 for inspiration.

Reply:
Yep, it's much better, thanks.